controllers/remove_services.go

```diff
@@ -34,26 +36,22 @@ type RemoveServices struct{}
 
 // Reconcile runs the reconciler's work.
 func (u RemoveServices) Reconcile(r *FoundationDBClusterReconciler, context ctx.Context, cluster *fdbtypes.FoundationDBCluster) (bool, error) {
-	service, err := GetHeadlessService(cluster)
-	if err != nil {
-		return false, err
-	}
-
-	if service != nil {
+	if GetHeadlessService(cluster) != nil {
 		return true, nil
 	}
 
-	existingServices := &corev1.ServiceList{}
-	err = r.List(context, existingServices, client.InNamespace(cluster.Namespace), client.MatchingField("metadata.name", cluster.Name))
+	existingService := &corev1.Service{}
+	err := r.Get(context, client.ObjectKey{Namespace: cluster.Namespace, Name: cluster.Name}, existingService)
 	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return true, nil
+		}
 		return false, err
 	}
 
-	if len(existingServices.Items) > 0 {
-		err = r.Delete(context, &existingServices.Items[0])
-		if err != nil {
-			return false, err
-		}
+	err = r.Delete(context, existingService)
```

Comment:
I think we should additionally check if the current cluster is the owner otherwise we could delete a service that is not part of the cluster (and managed by the operator).

Reply:
We might want to look at that as part of #290, which proposes using match labels instead of owner references when checking a similar relationship on pods.